src/stdlib_terminal_colors.f90

```diff
@@ -0,0 +1,138 @@
+! SPDX-Identifier: MIT
+
+!> Terminal color and style escape sequences
+module stdlib_terminal_colors
+    use stdlib_kinds, only : i1 => int8
+    implicit none
+    private
+
+    public :: ansi_color
+    public :: style_reset, style_bold, style_dim, style_italic, style_underline, &
+        & style_blink, style_blink_fast, style_reverse, style_hidden, style_strikethrough
+    public :: fg_color_black, fg_color_red, fg_color_green, fg_color_yellow, fg_color_blue, &
+        & fg_color_magenta, fg_color_cyan, fg_color_white, fg_color_default
+    public :: bg_color_black, bg_color_red, bg_color_green, bg_color_yellow, bg_color_blue, &
+        & bg_color_magenta, bg_color_cyan, bg_color_white, bg_color_default
+
+    public :: to_string, operator(+), operator(//)
+
+
+
+    !> Container for terminal escape code
+    type :: ansi_color
+        private
```

Comment:
There is one issue with intel (and it seems it follows the standard 2008), you can either change this to `public`
or change the lines where the error occurred to initialize each component individually

```
/Users/runner/work/stdlib/stdlib/src/stdlib_terminal_colors_operator.f90(19): error #6053: Structure constructor may not have components with the PRIVATE attribute   [STYLE]
            style=merge(rval%style, lval%style, rval%style >= 0), &
------------^
/Users/runner/work/stdlib/stdlib/src/stdlib_terminal_colors_operator.f90(20): error #6053: Structure constructor may not have components with the PRIVATE attribute   [FG]
            fg=merge(rval%fg, lval%fg, rval%fg >= 0), &
------------^
/Users/runner/work/stdlib/stdlib/src/stdlib_terminal_colors_operator.f90(21): error #6053: Structure constructor may not have components with the PRIVATE attribute   [BG]
            bg=merge(rval%bg, lval%bg, rval%bg >= 0))
```

Reply:
```suggestion
```